src/plugins/controls/public/react_controls/control_group/get_control_group_factory.tsx

```diff
@@ -133,7 +133,7 @@ export const getControlGroupEmbeddableFactory = (services: {
               controlsManager.api.children$
             ),
             controlGroupFetch$(ignoreParentSettings$, parentApi ? parentApi : {})
-          ),
+          ).pipe(distinctUntilChanged((a, b) => fastIsEqual(a, b))),
```

Comment:
Observable consumers should not need to test for distinct values. Instead, observables should only fire when they have new values. We should investigate which observable is emitting and prevent the emit with a duplicated value to resolve this issue

Reply:
That's fair - i have a feeling it is either the `query` or the `filters` where we aren't comparing objects properly. I'll dig into it.